engine/src/main/java/org/terasology/rendering/dag/nodes/FinalPostProcessingNode.java

```diff
@@ -53,50 +61,60 @@
     private DisplayResolutionDependentFBOs displayResolutionDependentFBOs;
 
     private RenderingDebugConfig renderingDebugConfig;
-    private RenderingConfig renderingConfig;
-    private FBO.Dimensions fullScale;
-    private Material finalPost;
-    private Material debug;
+    private EnableMaterial enablePostMaterial;
+    private EnableMaterial enableDebugMaterial;
 
+    /**
+     * This method must be called once shortly after instantiation to fully initialize the node
+     * and make it ready for rendering.
+     */
     @Override
     public void initialise() {
-        renderingConfig = config.getRendering();
-        renderingDebugConfig = renderingConfig.getDebug();
-        finalPost = worldRenderer.getMaterial("engine:prog.post"); // TODO: rename shader to finalPost
-        debug = worldRenderer.getMaterial("engine:prog.debug");
-        addDesiredStateChange(new BindFBO(FINAL));
+        renderingDebugConfig = config.getRendering().getDebug();
+        renderingDebugConfig.subscribe(RenderingDebugConfig.ENABLED, this);
+
+        enablePostMaterial = new EnableMaterial(POST_MATERIAL.toString());
```

Comment:
Yes, that's a change I intend to make. Right now I'm just refactoring all nodes with I have already available. Once that's done these kind of changes will be done across all nodes, all at once, in specific commits.